llmc/adamw.cuh

```diff
@@ -40,9 +40,7 @@ __device__ void adamw_update(Tp* params_memory, float* master_params_memory, Tg*
     float param = old_param - (learning_rate * (m / (sqrtf(v) + eps) + weight_decay * old_param));
     // update our low precision version of the parameters using stochastic rounding
     // this will be used in the next forward pass
-    // TODO: simply doing `params_memory[i] = (floatX)param;` breaks everything (why?)
-    unsigned int random = Get2dNoiseUint(threadIdx.x, blockIdx.x + blockDim.y * gridDim.y, seed);
-    stochastic_rounding(param, &params_memory[idx], random);
+    stochastic_rounding(param, &params_memory[idx], seed);
```

Comment:
isn't that destroying determinism, because now our  rounding depends on which address we get in memory?

Reply:
I don't think it is? We save the seed inside our state?

As for indexing the same threads always take care of the same params?

Reply:
sorry, yes, I was thinking about  `Get2dNoiseUint` where the second argument is part of the seed. Here it is fine.

Reply:
That's my second part of the comment - i don't think the PR as such breaks determinism? Because the indexing i pass into `Get2dNoiseUint` is deterministic.